src/types/python/excelsheet/mod.rs

```diff
@@ -574,6 +601,39 @@ impl ExcelSheet {
             .and_then(|obj| obj.into_bound_py_any(py))
     }
 
+    pub fn to_arrow_with_errors<'py>(&self, py: Python<'py>) -> PyResult<Bound<'py, PyAny>> {
+        let offset = self.offset();
+        let limit = self.limit();
+
+        let res = record_batch_from_data_and_columns_with_errors(
+            &self.selected_columns,
+            self.data(),
+            offset,
+            limit,
+        )
+        .with_context(|| {
+            format!(
+                "could not create RecordBatch from sheet \"{}\"",
+                self.name()
+            )
+        })?;
+
+        let rb_and_errors = (
+            res.0
+                .to_pyarrow(py)
+                .map_err(|err| FastExcelErrorKind::ArrowError(err.to_string()).into())
+                .with_context(|| {
+                    format!(
+                        "could not convert RecordBatch to pyarrow for sheet \"{}\"",
+                        self.name()
+                    )
+                })?,
+            res.1,
+        )
+            .into_bound_py_any(py)?;
+        Ok(rb_and_errors)
```

Comment:
Here, this would allow to simplify the code as:

```suggestion
        let rb_py = rb
            .to_pyarrow(py)
            .map_err(|err| FastExcelErrorKind::ArrowError(err.to_string()).into())
            .with_context(|| {
                format!(
                    "could not convert RecordBatch to pyarrow for sheet \"{}\"",
                    self.name()
                )
            })?
            .into_bound_py_any(py)?;
        Ok((rb_py, errors))
```

Reply:
That's no longer compatible with the current return type. Do you want me to change the return type as well and to what?

Reply:
Oops my bad, the last lines in this example should be:

```rust
         .with_context(|| {
                format!(
                    "could not convert RecordBatch to pyarrow for sheet \"{}\"",
                    self.name()
                )
            })?;
        Ok((rb_py, errors).into_bound_py_any(py)?)
```

Leave the return type as it is, this was just a suggestion to access the tuple fields.

In any case, the comment was intended as a Rust tip, I'm fine with the current form as well :slightly_smiling_face:

Reply:
Thanks, I appreciate the review/suggestions. I've updated it.